JSONPointer.java

```diff
@@ -37,8 +38,34 @@ of this software and associated documentation files (the "Software"), to deal
  * <a href="https://tools.ietf.org/html/rfc6901">RFC 6901</a>. 
  */
 public class JSONPointer {
+    
+    private static final String ENCODING = "utf-8";
```

Comment:
when we drop Java6 support, this can be changed to https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html

java.nio.charset.StandardCharsets.UTF_8

maybe put in a comment noting that.